### PR TITLE
switch to crio

### PR DESCRIPTION
**Pull request by sjenning**
@abhinavdahiya @mrunalp @derekwaynecarr @rphillips

xref https://github.com/openshift/machine-config-operator/pull/62

### CODE REVIEW

File: modules/ignition/assets.tf [hunk 1]
```diff
@@ -16,3 +16,13 @@ data "ignition_systemd_unit" "kubelet" {
...
+
+data "ignition_file" "sysconfig_crio_network" {
+  filesystem = "root"
+  mode       = "0644"
+  path       = "/etc/sysconfig/crio-network"
+
+  content {
+    content = "${file("${path.module}/resources/files/crio-network")}"
+  }
+}
```
[the remark this answers is not in the document]
Comment (sjenning) [reply modules/ignition/assets.tf]: resolved

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/machine-config-operator/pull/62 and https://github.com/openshift/machine-config-operator/pull/63 merged; now we wait for newest images to be published.

File: modules/ignition/resources/services/kubelet.service [hunk 2]
```diff
@@ -5,15 +5,17 @@ Wants=rpc-statd.service
...
   kubelet \
     --bootstrap-kubeconfig=/etc/kubernetes/kubeconfig \
     --kubeconfig=/var/lib/kubelet/kubeconfig \
     --rotate-certificates \
-    --cni-conf-dir=/etc/kubernetes/cni/net.d \
-    --cni-bin-dir=/var/lib/cni/bin \
-    --network-plugin=cni \
+    --container-runtime=remote \
+    --container-runtime-endpoint=unix:///var/run/crio/crio.sock \
+    --runtime-request-timeout=${KUBELET_RUNTIME_REQUEST_TIMEOUT} \
```
Comment (abhinavdahiya) [inline modules/ignition/resources/services/kubelet.service]: needs `$$` for terraform to skip trying to template this.

**Comment by abhinavdahiya** [comment]
testing locally:
```console
$ oc get node,pods -owide --all-namespaces
NAME                       STATUS    ROLES       AGE       VERSION           EXTERNAL-IP   OS-IMAGE                  KERNEL-VERSION               CONTAINER-RUNTIME
node/adahiya-0-bootstrap   Ready     bootstrap   10m       v1.11.0+d4cacc0   <none>        Red Hat CoreOS 4.0.5470   3.10.0-862.14.1.el7.x86_64   cri-o://1.11.2
node/adahiya-0-master-0    Ready     master      10m       v1.11.0+d4cacc0   <none>        Red Hat CoreOS 4.0.5470   3.10.0-862.14.1.el7.x86_64   cri-o://1.11.2
node/worker-246h5          Ready     worker      3m        v1.11.0+d4cacc0   <none>        Red Hat CoreOS 4.0.5470   3.10.0-862.14.1.el7.x86_64   cri-o://1.11.2
node/worker-r7xgd          Ready     worker      3m        v1.11.0+d4cacc0   <none>        Red Hat CoreOS 4.0.5470   3.10.0-862.14.1.el7.x86_64   cri-o://1.11.2

NAMESPACE                           NAME                                                        READY     STATUS    RESTARTS   AGE       IP               NODE
default                             pod/registry-846b59554f-8x4ht                               1/1       Running   0          4m        10.2.2.3         worker-r7xgd
kube-system                         pod/kube-apiserver-bnhrd                                    1/1       Running   0          10m       192.168.124.11   adahiya-0-master-0
kube-system                         pod/kube-controller-manager-b6985978d-fcdq5                 1/1       Running   0          10m       10.2.1.5         adahiya-0-master-0
kube-system                         pod/kube-core-operator-854857854d-5xc9l                     1/1       Running   0          5m        10.2.1.13        adahiya-0-master-0
kube-system                         pod/kube-dns-787c975867-d2562                               3/3       Running   0          10m       10.2.1.4         adahiya-0-master-0
kube-system                         pod/kube-flannel-4r9zs                                      2/2       Running   1          3m        192.168.124.51   worker-r7xgd
kube-system                         pod/kube-flannel-7pxw9                                      2/2       Running   0          7m        192.168.124.10   adahiya-0-bootstrap
kube-system                         pod/kube-flannel-cxd4h                                      2/2       Running   0          7m        192.168.124.11   adahiya-0-master-0
kube-system                         pod/kube-flannel-gw6fg                                      2/2       Running   0          3m        192.168.124.52   worker-246h5
kube-system                         pod/kube-proxy-d8fsz                                        1/1       Running   0          10m       192.168.124.10   adahiya-0-bootstrap
kube-system                         pod/kube-proxy-h9s8k                                        1/1       Running   0          3m        192.168.124.51   worker-r7xgd
kube-system                         pod/kube-proxy-jlq57                                        1/1       Running   0          3m        192.168.124.52   worker-246h5
kube-system                         pod/kube-proxy-r4l7c                                        1/1       Running   0          10m       192.168.124.11   adahiya-0-master-0
kube-system                         pod/kube-scheduler-78d86f9754-vc4sm                         1/1       Running   0          10m       10.2.1.2         adahiya-0-master-0
kube-system                         pod/machine-api-operator-757575d7c4-mw6xv                   1/1       Running   0          10m       10.2.1.7         adahiya-0-master-0
kube-system                         pod/metrics-server-5767bfc576-snvsm                         2/2       Running   0          4m        10.2.2.5         worker-r7xgd
kube-system                         pod/openshift-apiserver-nbm9q                               1/1       Running   0          10m       192.168.124.11   adahiya-0-master-0
kube-system                         pod/openshift-controller-manager-d9787b9c-m9grc             1/1       Running   0          9m        10.2.1.3         adahiya-0-master-0
kube-system                         pod/pod-checkpointer-hdr69                                  1/1       Running   0          10m       192.168.124.11   adahiya-0-master-0
kube-system                         pod/pod-checkpointer-hdr69-adahiya-0-master-0               1/1       Running   0          8m        192.168.124.11   adahiya-0-master-0
kube-system                         pod/tectonic-network-operator-c4b2k                         1/1       Running   0          10m       192.168.124.11   adahiya-0-master-0
openshift-cluster-api               pod/clusterapi-apiserver-79997c8756-sf9xx                   2/2       Running   0          5m        10.2.1.9         adahiya-0-master-0
openshift-cluster-api               pod/clusterapi-controllers-85f6bfd9d5-hxw7w                 2/2       Running   0          5m        10.2.1.14        adahiya-0-master-0
openshift-ingress                   pod/default-http-backend-6985d557bb-q7znx                   1/1       Running   0          4m        10.2.2.8         worker-r7xgd
openshift-ingress                   pod/router-6cd5cfb9f8-8kjwj                                 1/1       Running   0          3m        10.2.2.6         worker-r7xgd
openshift-ingress                   pod/tectonic-ingress-controller-operator-599fdd5cff-rpjnb   1/1       Running   0          5m        10.2.1.15        adahiya-0-master-0
openshift-machine-config-operator   pod/machine-config-controller-67d496b48d-268m4              1/1       Running   0          5m        10.2.1.16        adahiya-0-master-0
openshift-machine-config-operator   pod/machine-config-daemon-bw4vz                             1/1       Running   0          3m        10.2.2.7         worker-r7xgd
openshift-machine-config-operator   pod/machine-config-daemon-js96c                             1/1       Running   0          3m        10.2.3.2         worker-246h5
openshift-machine-config-operator   pod/machine-config-daemon-lcjgg                             1/1       Running   0          3m        10.2.1.21        adahiya-0-master-0
openshift-machine-config-operator   pod/machine-config-operator-776b444b99-wrppk                1/1       Running   0          9m        10.2.1.6         adahiya-0-master-0
openshift-machine-config-operator   pod/machine-config-server-dc9jw                             1/1       Running   0          4m        192.168.124.11   adahiya-0-master-0
openshift-web-console               pod/webconsole-6698d4fbbc-k6rpd                             1/1       Running   0          4m        10.2.2.2         worker-r7xgd
tectonic-system                     pod/directory-sync-d84d84d9f-5gfgr                          1/1       Running   0          4m        10.2.2.4         worker-r7xgd
tectonic-system                     pod/kube-addon-operator-775d4c8f8d-pvb8z                    1/1       Running   0          5m        10.2.1.10        adahiya-0-master-0
tectonic-system                     pod/tectonic-alm-operator-79b6996f74-srrpl                  1/1       Running   0          5m        10.2.1.8         adahiya-0-master-0
tectonic-system                     pod/tectonic-channel-operator-6d9d4fd8ff-t67mr              1/1       Running   0          5m        10.2.1.20        adahiya-0-master-0
tectonic-system                     pod/tectonic-clu-6b8d87785f-clwjs                           1/1       Running   0          4m        10.2.1.19        adahiya-0-master-0
tectonic-system                     pod/tectonic-utility-operator-78dfcdb9dd-xwcd9              1/1       Running   0          5m        10.2.1.12        adahiya-0-master-0
```

networking across nodes:
```console
$ kubectl run -it --rm --restart=Never busybox-curl --image=radial/busyboxplus:curl sh
$ curl -k https://kubernetes.default.svc.cluster.local:443/api
{
  "kind": "APIVersions",
  "versions": [
    "v1"
  ],
  "serverAddressByClientCIDRs": [
    {
      "clientCIDR": "0.0.0.0/0",
      "serverAddress": "192.168.124.11:6443"
    }
  ]
}
```

**Comment by wking** [comment]
We're back to an earlier release config with openshift/release#1518. Kicking again:

/retest

**Pull request merged**